test/replicache_test.dart

```diff
@@ -657,4 +661,69 @@ Future<void> main() async {
     expect(deleteCount, 4);
     expect(createCount, 3);
   });
+
+  test('nested tx', () async {
+    await useReplay('nested tx');
+
+    rep = await replicacheForTesting('nested-tx');
+
+    expect(await rep.query((tx) async => 'abc'), 'abc');
+
+    expect(
+      rep.query((tx) async {
+        await rep.query((tx) => null);
+      }),
+      throwsNestedTransactionError,
+    );
+
+    final c = Completer();
+    var err;
+
+    Future<void> tricky() async {
+      try {
+        await rep.query((tx) => null);
+      } catch (e) {
+        err = e;
+        c.complete();
+      }
+    }
+
+    await rep.query((tx) async {
+      Timer(Duration(milliseconds: 10), tricky);
+    });
+
+    await c.future;
+    expect(err, isA<NestedTransactionError>());
+  });
+
+  test('nested tx mutate', () async {
+    await useReplay('nested tx mutate');
+
+    rep = await replicacheForTesting('nested-tx-mutate');
+
+    final mut = rep.register('mut', (tx, args) async {
+      return args;
+    });
+    expect(await mut(42), 42);
+
+    final mut2 = rep.register('mut2', (tx, args) async {
+      if (args == 0) {
+        return 0;
+      }
+      return (await mut(args - 1)) + 1;
+    });
+    expect(mut2(1), throwsNestedTransactionError);
+
+    final mut3 = rep.register('mut3', (tx, args) => rep.get('abc'));
+    expect(mut3(null), throwsNestedTransactionError);
+  });
+
+  test('nested tx subscribe', () async {
+    await useReplay('nested tx subscribe');
+
+    rep = await replicacheForTesting('nested-tx-subscribe');
+
+    final s = rep.subscribe((tx) => rep.get('abc'));
```

Comment:
i don't see the nesting here, why does this throw?

Reply:
subscribe opens a tx (using query). The body calls `rep.get` which calls `rep.query`.

I wanted this test to ensure that the stream contains the error.